xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-profile/xwiki-platform-user-profile-ui/src/main/resources/XWiki/XWikiUserProfileSheet.xml

```diff
@@ -141,6 +141,27 @@
       <code>require(['jquery', 'xwiki-meta'], function ($, xm) {
   'use strict';
   $(document).ready(function() {
+
+    /** Add autocomplete attributes to ease form filling. */
+    const prefix = 'XWiki.XWikiUsers_0_';
+    var fieldsPurposes = {
+      'first_name': 'given-name',
```

Comment:
quite frankly I'm not a big fan of this structure, I'd have preferred something such as: 
```
var fieldsPurposes = [
  {
    'name': 'first_name',
    'purpose': 'given-name'
  },
  ...
]
```
which is more explicit and allows extension IMO. It also helps to understand the code below since you'd use directly `entry.name` and `entry.purpose`.

Reply:
Indeed, thanks, I used this structure in the latest commit.